From mboxrd@z Thu Jan  1 00:00:00 1970
Return-Path: <linux-kernel-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 3A146C433F5
	for <linux-kernel@archiver.kernel.org>; Tue,  9 Nov 2021 09:49:58 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 22A06611CC
	for <linux-kernel@archiver.kernel.org>; Tue,  9 Nov 2021 09:49:58 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S245048AbhKIJwl (ORCPT <rfc822;linux-kernel@archiver.kernel.org>);
        Tue, 9 Nov 2021 04:52:41 -0500
Received: from lindbergh.monkeyblade.net ([23.128.96.19]:49294 "EHLO
        lindbergh.monkeyblade.net" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S244164AbhKIJvj (ORCPT
        <rfc822;linux-kernel@vger.kernel.org>);
        Tue, 9 Nov 2021 04:51:39 -0500
Received: from mail-pl1-x636.google.com (mail-pl1-x636.google.com [IPv6:2607:f8b0:4864:20::636])
        by lindbergh.monkeyblade.net (Postfix) with ESMTPS id B410CC061764
        for <linux-kernel@vger.kernel.org>; Tue,  9 Nov 2021 01:48:53 -0800 (PST)
Received: by mail-pl1-x636.google.com with SMTP id y1so19905343plk.10
        for <linux-kernel@vger.kernel.org>; Tue, 09 Nov 2021 01:48:53 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=sifive.com; s=google;
        h=from:to:subject:date:message-id:in-reply-to:references:mime-version
         :content-transfer-encoding;
        bh=P9b3tDY+655xD0tPu9SBTJCGpaBZsqsfT1tLH+eh9ss=;
        b=lH+Zi+btz0Y3PlO5Le6vD8gIS2gNoRc+CDGfvwK17wjnrtv9A40qM8mlBQvpTgMzfv
         eiKZw0ZQTiDbMwH/xUeRYTyag0FYNP1oWYpOmMyM2yiXFnw4/HcDC7w1AGojiPJ3Fn5y
         Fuw3/MPUiXHruI1mYFcflj+FDpE4U7yI4DIoQmV/hMk7+y/+ZKTCMkV8eJh1AuYi0Tmm
         i12FqNhrilpzfcYiwyAt39z+QKkiVw6/n0bdhmBtJwQ+rYaOvvQZaPrlitURm795ecsU
         740yWzE/PYiFOuRFFyFCCRRZaNekgfQQWfkm+t8fLYq+tL2U1mau/YYPW3T1DID2M/XP
         UGYQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20210112;
        h=x-gm-message-state:from:to:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=P9b3tDY+655xD0tPu9SBTJCGpaBZsqsfT1tLH+eh9ss=;
        b=mlbAK3ahA0azrlu5qybtnarDAlAMWf0XvAkrxCOkJu9G2XakPjyfu+I0GxshdGbdgZ
         mnjvdmmO+YV687Q7OKQFzwdSB5o7XTgZ4fNpDrrXimyLHVslQFwh/QO0DIwGnYS9aLVy
         TH2WAzdWPX9vElU0gWQTofgqaJODAaTNjIUkCV3LwlJd/rM2y51wRmYKsaTA8PirObRC
         P8nB4DK9td7zN8VgNlKj6S8U/e20f/se6lf2jYK21JlVC1yqjTv+pX75ESvYfJGPyHAG
         K0fSyZAjry9UQTLdBQgWHRQjkhWpeMOJ1sqDUCfuj5QVzsBXfDFyvp4zfINzh0ovtotN
         dAnw==
X-Gm-Message-State: AOAM531u2M1n9hqDoQjmpon9Wcj3NREhw0sWko5UQYZqAy6aEmnml1Bs
        LzSAsZS3lPPebc02L2b3/M774A==
X-Google-Smtp-Source: ABdhPJyE89/bOjt3ujPXxTBFVPDjVmC+oQVF+9ESGIp8HQlLll/3ZSvGbSTYp6vIWSHhcnEsAw8slQ==
X-Received: by 2002:a17:90a:9291:: with SMTP id n17mr5695519pjo.243.1636451333257;
        Tue, 09 Nov 2021 01:48:53 -0800 (PST)
Received: from hsinchu16.internal.sifive.com (59-124-168-89.hinet-ip.hinet.net. [59.124.168.89])
        by smtp.gmail.com with ESMTPSA id cv1sm1940511pjb.48.2021.11.09.01.48.52
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Tue, 09 Nov 2021 01:48:53 -0800 (PST)
From:   Greentime Hu <greentime.hu@sifive.com>
To:     palmer@dabbelt.com, paul.walmsley@sifive.com,
        linux-riscv@lists.infradead.org, linux-kernel@vger.kernel.org,
        aou@eecs.berkeley.edu
Subject: [PATCH v9 14/17] riscv: Use CSR_STATUS to replace sstatus in vector.S
Date:   Tue,  9 Nov 2021 17:48:26 +0800
Message-Id: <7204695c80a13a634381fed94ea1598ffd70c78f.1636362169.git.greentime.hu@sifive.com>
X-Mailer: git-send-email 2.31.1
In-Reply-To: <cover.1636362169.git.greentime.hu@sifive.com>
References: <cover.1636362169.git.greentime.hu@sifive.com>
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
Precedence: bulk
List-ID: <linux-kernel.vger.kernel.org>
X-Mailing-List: linux-kernel@vger.kernel.org

It should use the same logic here in both m-mode and s-mode.

Signed-off-by: Greentime Hu <greentime.hu@sifive.com>
Co-developed-by: Vincent Chen <vincent.chen@sifive.com>
Signed-off-by: Vincent Chen <vincent.chen@sifive.com>
---
 arch/riscv/kernel/vector.S | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/riscv/kernel/vector.S b/arch/riscv/kernel/vector.S
index a973532d2e7f..1a79e2485bfa 100644
--- a/arch/riscv/kernel/vector.S
+++ b/arch/riscv/kernel/vector.S
@@ -31,7 +31,7 @@
 
 ENTRY(__vstate_save)
 	li      status, SR_VS
-	csrs    sstatus, status
+	csrs    CSR_STATUS, status
 
 	csrr    x_vstart, CSR_VSTART
 	csrr    x_vtype, CSR_VTYPE
@@ -51,13 +51,13 @@ ENTRY(__vstate_save)
 	REG_S   x_vl, RISCV_V_STATE_VL(vstatep)
 	REG_S   x_vcsr, RISCV_V_STATE_VCSR(vstatep)
 
-	csrc	sstatus, status
+	csrc	CSR_STATUS, status
 	ret
 ENDPROC(__vstate_save)
 
 ENTRY(__vstate_restore)
 	li      status, SR_VS
-	csrs    sstatus, status
+	csrs    CSR_STATUS, status
 
 	vsetvli incr, x0, e8, m8
 	vle8.v   v0, (datap)
@@ -76,7 +76,7 @@ ENTRY(__vstate_restore)
 	csrw    CSR_VSTART, x_vstart
 	csrw    CSR_VCSR, x_vcsr
 
-	csrc	sstatus, status
+	csrc	CSR_STATUS, status
 	ret
 ENDPROC(__vstate_restore)
 
-- 
2.31.1


